lib/elixir/pages/anti-patterns/what-anti-patterns.md

```diff
@@ -0,0 +1,39 @@
+# What are anti-patterns?
+
+Anti-patterns describe common mistakes or indicators of potential problems in code.
+They are also known by the term "code smells".
+
+The goal of these guides is to document known anti-patterns found in Elixir software
+and teach developers how to identify and correct them. If an existing piece of code
+matches an anti-pattern, it does not mean your code must be rewritten. However, you
+should take its potential pitfalls and alternatives into consideration.
+
+The anti-patterns in these guides are broken into 4 main categories:
+
+  * **Code-related anti-patterns:** related to your code and particular
+    language idioms and features;
+
+  * **Design-related anti-patterns:** related to your modules, functions,
+    and the role they play within a codebase;
+
```

Comment:
Maybe a rephrasing, wdyt of this:

```suggestion
  * **Design-related anti-patterns:** related to the design of your application, including how modules and functions work with each other within a codebase;

```

Reply:
I like that BUT I wanted to avoid using the term "application". For example, maybe some will think it does not apply to a library. Thoughts?

Reply:
You can replace "application" with "code" in the proposed sentence and I still think it makes sense 👍

Reply:
The problem is that it then starts to sound just like "code-related anti-patterns".